src/material-experimental/selection/selection-column.ts

```diff
@@ -0,0 +1,102 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {MatCellDef, MatColumnDef, MatHeaderCellDef, MatTable} from '@angular/material/table';
+import {
+  Component,
+  Input,
+  isDevMode,
+  OnDestroy,
+  OnInit,
+  Optional,
+  ViewChild,
+  ChangeDetectionStrategy,
+  ViewEncapsulation,
+} from '@angular/core';
+
+import {MatSelection} from './selection';
+
+/**
+ * Column that adds row selecting checkboxes and a select-all checkbox if `matSelectionMultiple` is
+ * `true`.
+ *
+ * Must be used within a parent `MatSelection` directive.
+ */
+@Component({
+  selector: 'mat-selection-column',
+  template: `
+    <ng-container matColumnDef>
+      <th matHeaderCell *matHeaderCellDef align="center">
+        <mat-checkbox *ngIf="selection.multiple"
+            matSelectAll
+            #allToggler="matSelectAll"
+            [indeterminate]="allToggler.indeterminate | async"></mat-checkbox>
+      </th>
+      <td matCell *matCellDef="let row; let i = $index" align="center">
+        <mat-checkbox
+            matSelectionToggle
+            [matSelectionToggleValue]="row"
+            [matSelectionToggleIndex]="i"></mat-checkbox>
+      </td>
+    </ng-container>
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+})
+export class MatSelectionColumn<T> implements OnInit, OnDestroy {
+  /** Column name that should be used to reference this column. */
+  @Input('matSelectionColumnName')
```

Comment:
The alias isn't necessary here since it's a custom element, it can just be `name`

Reply:
Ah yes. I will send another PR to correct the CDK files too.